### PR TITLE
Fix PostGetPutGetDeleteGet Test

### PR DESCRIPTION
**Pull request by bean00**
For `PostGetPutGetDeleteGet`, the first `GET` should actually fail, because the `form` file wasn't in the `public` directory. There were other issues as well.

With the new `CreateReadUpdateDelete` test, these issues are addressed. The `Location` header has been added in as a requirement as well (see https://tools.ietf.org/html/rfc7231#section-4.3.3):

> If one or more resources has been created on the origin server as a
   result of successfully processing a POST request, the origin server
   SHOULD send a 201 (Created) response containing a Location header
   field that provides an identifier for the primary resource created
   (Section 7.1.2) and a representation that describes the status of the
   request while referring to the new resource(s).

The specification that isn't being met is including a description of the status of the request in the response body for the `POST` request.

### CODE REVIEW

**Comment by damonkelley** [comment]
I am not sure if I agree that this spec is broken. I understand what you mean about `POST` responding with a `201` when the resource is created, but the nature of the spec indicates that the `form` resource already exists, which is valid. 

Some of Cob Spec is dealing with the filesystem (specifically files in the public directory), but I don't think that all resources are intended map to a file necessarily, this `form` resource is an example of that.

I am open to hearing more about this though.

**Comment by jdesrosiers** [comment]
This is arguably the worst test in cob-spec. It's long past time for an overhaul. @damonkelley you're right that not all resources need to live on the filesystem, but there are other problems with this test as well.

My biggest issue with it is that the `/form` resource is an angel resource. When you `DELETE` it, it miraculously respawns immediately. If a resource is `DELETE`d, the resource is expected to be gone. Any subsequent request should result in 404 no matter which HTTP method is used. Technically the server could do whatever it wants, but this behavior is quite unexpected. The only way I can see to fix this problem is to separate the creator resource from the resource it creates as is done in this PR.

The original test is at best confusing and bad practice. The new test on the other hand models the most common pattern used in REST APIs and it only required a relatively small change. I think this is a big win.

File: FitNesseRoot/HttpTestSuite/ResponseTestSuite/CreateReadUpdateDelete/content.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+|put       |/form/data                                         |
+|ensure    |response code equals              |200             |
+|get       |/form/data                                         |
+|ensure    |response code equals              |200             |
+|ensure    |body has content                  |data=heathcliff |
+|delete    |/form/data                                         |
+|ensure    |response code equals              |200             |
+|get       |/form/data                                         |
+|ensure    |response code equals              |404             |
+|delete    |/form                                              |
```
Comment (jdesrosiers) [inline FitNesseRoot/HttpTestSuite/ResponseTestSuite/CreateReadUpdateDelete/content.txt]: The `/form` should not be `DELETE`d. Not unless you don't expect anyone to be able to `POST` to in again. I think it makes most sense to add the `/from` directory to the public folder.

**Comment by tbaik** [comment]
One more problem this fixes is also with the confusing use of POST/PUT on `/form`. I think in practice, POSTing on `/form` is usually done to create or update on its child resource (like data) and PUT would be done on `/form/data`, the child resource

**Comment by bean00** [comment]
**Note**: I will rebase all the commits for this PR together before merging.

**Comment by damonkelley** [comment]
@jdesrosiers Thanks for the clarification!

**Comment by bean00** [comment]
Awesome, thanks @damonkelley  for merging in the PR!

**Pull request merged**